### PR TITLE
Enhancement/21 display node monitoring result

### PR DESCRIPTION
**Pull request by h0n9**
resolves https://github.com/amolabs/explorer2/issues/21

### CODE REVIEW

File: server/models/incentive.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* vim: set sw=2 ts=2 expandtab : */
+const db = require('../db/db');
+
+async function getOne(chain_id, address, from, num) {
```
Comment (Lbird) [inline server/models/incentive.js:4]: weird name. why get`One` returns list? rather change it to `getListByAddress`.

File: server/models/incentive.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      }
+      return resolve(rows);
+    });
+  });
+}
+
+module.exports = {
+  getOne: getOne,
+  getList: getList,
+}
```
Comment (Lbird) [inline server/models/incentive.js:41]: fix this

File: server/models/node.js [hunk 3]
```diff
@@ -1,62 +1,85 @@
 /* vim: set sw=2 ts=2 expandtab : */
 const db = require('../db/db');
 
-async function getOne(chain_id, node_id) {
+async function getOne(chain_id, val_addr, from, to) {
```
Comment (Lbird) [inline server/models/node.js:4]: Here again. why `getOne` returns list?

File: server/models/node.js [hunk 4]
```diff
@@ -1,62 +1,85 @@
...
+  JOIN ( \
+    SELECT `node_id`, MAX(`timestamp`) `max_timestamp` \
+    FROM `node_info` \
+    WHERE `chain_id` = ? AND `val_addr` = ? AND `timestamp` BETWEEN ? AND ? \
+    GROUP BY `node_id` \
+  ) mt \
+  ON n.`node_id` = mt.`node_id` AND ni.`timestamp` = mt.`max_timestamp` \
+  ORDER BY n.`moniker`";
+    query_var = [chain_id, val_addr, from, to, from, to, chain_id, val_addr, from, to];
+    db.query(query_str, query_var, function(err, row, fields) {
```
Comment (Lbird) [inline server/models/node.js:39]: Too complex query. I don't get what it means. leave a comment explaining this query.

File: server/models/node.js [hunk 5]
```diff
@@ -1,62 +1,85 @@
...
         if(err) {
             return reject(err);
         }
-        return resolve(rows[0]);
+        return resolve(row);
```
Comment (Lbird) [inline server/models/node.js:43]: Strange. does this query return multiple or single?

File: server/models/node.js [hunk 6]
```diff
@@ -1,62 +1,85 @@
...
-            AND i.`node_id` = m.`node_id` \
-            AND i.`latest_block_height` = m.`latest_block_height` \
-          WHERE n.`chain_id` = m.`chain_id` AND n.`node_id` = m.`node_id` \
-          ORDER BY n.`moniker` \
-          LIMIT ?, ?";
-        var query_var = [chain_id, from, num];
+            ) mt \
+            ON n.`node_id` = mt.`node_id` AND ni.`timestamp` = mt.`max_timestamp` \
+            ORDER BY n.`moniker`";
+        var query_var = [chain_id, from, to, from, to, chain_id, from, to];
```
Comment (Lbird) [inline server/models/node.js:82]: Again. explain the query.

File: server/models/node.js [hunk 7]
```diff
@@ -66,40 +89,37 @@ async function getList(chain_id, from, num) {
     });
 }
 
-async function getHistory(chain_id, node_id, anchor, from, num) {
+async function getHistory(chain_id, val_addr, anchor, from, num) {
```
Comment (Lbird) [inline server/models/node.js:92]: What is the difference between this and `getOne` ? almost the same signature except `anchor`.

File: server/routes/nodes.js [hunk 8]
```diff
@@ -5,9 +5,27 @@ const node = require('../models/node');
...
+  to_timestamp = req.query.upto || to_timestamp;
+  var from = new Date(Date.parse(to_timestamp) - (range * 1000));
+  var from_timestamp = from.getUTCFullYear() + '-' 
+                     + (from.getUTCMonth()+1) + '-'
+                     + from.getUTCDate() + ' '
+                     + from.getUTCHours() +':'
+                     + from.getUTCMinutes() + ':'
+                     + from.getUTCSeconds();
+
+  console.log(from_timestamp, "~", to_timestamp);
```
Comment (Lbird) [inline server/routes/nodes.js:26]: comment out this

File: server/models/node.js (hunk 7 above)
Comment (h0n9) [reply server/models/node.js:92]: - `getOne`: latest status of a node given from client (single value)
- `getList`: latest status of nodes (list)
- `getHistory`: history status of a node given from client (list)

**Pull request merged**